### DISCUSSION TITLE
search german BnetzA Server in addittion to qrz.com if german HAM callsign entered

### DISCUSSION THREAD

**Discussion by fds2610** [magicbug/Cloudlog#2709] [Ideas / Feature Requests]
Hi,
I am new to ham radio, but in Germany i experience that many OMs do not update their QRZ.com entry when obtaining a new callsign, because our federal regulatory offers a webservice to search for actual callsign owners: https://ans.bundesnetzagentur.de/Amateurfunk/Rufzeichen.aspx
Is it possible to search in that database for actualized data instead of qry.com whenever a german callsign is entered ? Maybe other countries have similar (official) services to integrate?

**Comment by HB9HIL** [comment]
Good morning @fds2610 

First of all: Thanks for using Cloudlog. 

About BNA Search: I think this will be a difficult one and probably never happens to be implemented. Reason for that is that the whole lookup stuff, like qrz.com/hamqth is a grey zone because of DSGVO. And connecting with a Government Site which hasn't an API is critical here. There would be a way to get the results by using curl but only with a state token go around. In fact it's probably not 100% legally correct. 

We plan to add a plugin feature to Cloudlog so maybe one day we could offer something like this as Plugin but there is no ETA for this. 

73 HB9HIL

(and may some greetings to Göppingen) 😉

**Reply by fds2610** [reply]
Hi Fabian,

well I am not a lawyer nor am i educated in EU law, though i have a strong opinion on what is legal and what not - as maybe everyone ;-) It is sad to read that my suggestion maybe wont make it on a todo list as current qrz lookup really is much less helpful at least here in germany. I hoped that i do not need to lookup callsigns in in several sources.
As it is really an important nice-to-have-feature i would offer to contribute in realizing this feature. If there is anybody in this project who can do the programing stuff I would offer to get in contact with BNetzA officials and ask for official permission or find out under what circumstances automatic requests are allowed on that page. I reviewed the html already and it should not be too complicated to write a page-scraper to get the requested information even though there is no API provided.
regards , Felix

P.S. my QTH is Leonberg + many greets to Appenzell - or whereever your QTH is as your shack seems to be ultra-mobile ;-)

**Reply by HB9HIL** [reply]
Haha "ultra-mobile" fits the most to my circumstances. 

Stay alert for a Plugin-feature probably 2024 and come back to this discussion. So maybe we can address that as a Plugin. 👍
